HouseRobber.java

```diff
@@ -0,0 +1,20 @@
+// Time Complexity :O(n) --> beacuse of for loop
+// Space Complexity :O(1)  ---> because of Hashmap
+// Did this code successfully run on Leetcode : No
+// Any problem you faced while coding this : Didn'' understand how to approach for different test cases
+
+
+// Your code here along with comments explaining your approach
+
+class Solution {
+    public int rob(int[] nums) {
```

Comment:
Its not the case of adding even and odd.
You need to where to go (n+2 or n+3) whichever is maximum out of these two.

Reply:
Completed using dynamic programming approach by finding the maximum amount a robber can rob on reaching a particular house at index 'i' and adding them to a dp array . Finally return the last value of the array.